pkg/network/ebpf/c/runtime/conntrack.c

```diff
@@ -28,37 +28,52 @@ int kprobe___nf_conntrack_hash_insert(struct pt_regs* ctx) {
         increment_telemetry_count(registers_dropped);
         return 0;
     }
+    
+    log_debug("kprobe/__nf_conntrack_hash_insert: netns: %u, status: %x\n", get_netns(&ct->ct_net), status);
+
+    conntrack_tuple_t orig = {}, reply = {};
+    if (nf_conn_to_conntrack_tuples(ct, &orig, &reply) != 0) {
+        return 0;
+    }
 
-    struct nf_conntrack_tuple_hash tuplehash[IP_CT_DIR_MAX];
-    __builtin_memset(tuplehash, 0, sizeof(tuplehash));
-    bpf_probe_read_kernel(&tuplehash, sizeof(tuplehash), &ct->tuplehash);
+    bpf_map_update_elem(&conntrack, &orig, &reply, BPF_ANY);
+    bpf_map_update_elem(&conntrack, &reply, &orig, BPF_ANY);
+    increment_telemetry_count(registers);
+
+    return 0;
+}
 
-    struct nf_conntrack_tuple orig = tuplehash[IP_CT_DIR_ORIGINAL].tuple;
-    struct nf_conntrack_tuple reply = tuplehash[IP_CT_DIR_REPLY].tuple;
+SEC("kprobe/ctnetlink_fill_info")
+int kprobe_ctnetlink_fill_info(struct pt_regs* ctx) {
 
-    u32 netns = get_netns(&ct->ct_net);
-    log_debug("kprobe/__nf_conntrack_hash_insert: netns: %u, status: %x\n", netns, status);
+    proc_t proc = {};
+    bpf_get_current_comm(&proc.comm, sizeof(proc.comm));
 
-    conntrack_tuple_t orig_conn = {};
-    if (!nf_conntrack_tuple_to_conntrack_tuple(&orig_conn, &orig)) {
+    if (!proc_t_comm_prefix_equals("system-probe", 12, proc)) {
+        log_debug("skipping kprobe/ctnetlink_fill_info invocation from non-system-probe process\n");
         return 0;
     }
-    orig_conn.netns = netns;
 
-    log_debug("orig\n");
-    print_translation(&orig_conn);
+    struct nf_conn *ct = (struct nf_conn*)PT_REGS_PARM5(ctx);
 
-    conntrack_tuple_t reply_conn = {};
-    if (!nf_conntrack_tuple_to_conntrack_tuple(&reply_conn, &reply)) {
+    u32 status = ct_status(ct);
+    if (!(status&IPS_CONFIRMED)) {
+        return 0;
+    }
+    if (!(status&IPS_NAT_MASK)) {
+        increment_telemetry_count(registers_dropped);
```

Comment:
I know you are duplicating code which already exists in `__nf_conntrack_hash_insert`, but I reconsidering what this telemetry is even telling us. What do you think about removing the `registers_dropped` telemetry period? I think it was valuable for netlink, to have sense of the total volume. In eBPF, I don't think it is really telling us much.